components/Calendar/CalendarPage.jsx

```diff
@@ -0,0 +1,218 @@
+
+    this.state = this.defaultState;
+  }
+
+  componentDidMount() {
+    // TODO: remove this hack
+    this.isComponentMounted = true;
+    this.toggleOverflow(false);
+    this.fetchEvents();
+  }
+
+  componentWillUnmount() {
+    this.isComponentMounted = false;
+    this.toggleOverflow(false);
+  }
+
+  fetchEvents = () => {
+    this.resetState();
+
+    fetch('https://calendar-api.now.sh/')
+      .then(response => response.json())
+      .then(monthlyCalendars => {
+        if (this.isComponentMounted) {
+          this.setState({
+            error: false,
+            loading: false,
+            monthlyCalendars: monthlyCalendars.map(monthlyCalendar =>
+              Object.assign({}, monthlyCalendar, {
+                events: monthlyCalendar.events.map(event =>
+                  Object.assign({}, event, {
+                    shortid: shortid.generate()
+                  })
+                )
+              })
+            )
+          });
+        }
+      })
+      .catch(() => {
+        if (this.isComponentMounted) {
+          this.setState({
+            error: true,
+            loading: false
+          });
+        }
+      });
+  };
+
+  getFormattedEventHour = date => format(new Date(date).setUTCMinutes(180), 'HH:mm');
+
+  hideModal = () => {
+    this.setState({ showModal: false }, () => this.toggleOverflow(false));
+  };
+
+  resetState = () => {
+    this.setState(this.defaultState);
+  };
+
+  showModal = (eventsOfTheDay, currentDay) => {
+    this.setState(
+      {
+        currentDay,
+        eventsOfTheDay,
+        showModal: true
+      },
+      () => this.toggleOverflow(true)
+    );
+  };
+
+  toggleOverflow = active => {
+    // TODO: puedo lograr esto de una manera mejor? Modificar un elemento del DOM
+    // que no solo está fuera de mi componente sino que fuera del root de la app
+    document.querySelector('body').classList[active ? 'add' : 'remove']('overflow-hidden');
+  };
+
+  render() {
+    const { showOnlyCurrentMonth } = this.props;
+    const { currentDay, error, eventsOfTheDay, loading, monthlyCalendars, showModal } = this.state;
+    const monthlyCalendarsToShow = showOnlyCurrentMonth ? [monthlyCalendars[0]] : monthlyCalendars;
+
+    if (loading) {
+      return <Loading message="Buscando eventos..." />;
+    }
+
+    if (error) {
+      return (
+        <MessageWithAction
+          action={this.fetchEvents}
+          actionMessage="Intentar nuevamente"
+          image="/static/Calendar/fail.gif"
+          imageAlt="Error"
+          message="Ocurrió un error al traer los eventos."
+        />
+      );
+    }
+
+    return (
+      <React.Fragment>
+        <Container large="large">
+          <div className="fade-in">
+            <div className="flex items-center justify-center">
+              <h1 className="mv0">Calendario de eventos</h1>
+              <Link href="/articulos/que-es-el-calendario-de-eventos" passHref>
+                <a
+                  href="#!"
```

Comment:
Debugging the code with devTools I saw that without `passHref` the link doesn't have an `href` prop, so the browser doesn't suggest the link at the bottom as a regular link and I don't like it. And the `href="#!"` is to avoid linter errors